### PR TITLE
Add sample unit test

### PR DESCRIPTION
**Pull request by daichiro**
Related to #5 
Follow #8 

## Overview

- Add **sample** unit test

If this proposal is approved, I will create more test cases.

## For testing

- Usually we run tests, dependency injection (DI) is needed.

`QRScanner` uses `AVFoundation`'s classes.
They depend on running device situation.
So we should use mock of them.

### For mocking

I create `QRScanner.Input`.
This can inject dependency when initializing.

In future, I add injectable properties like `AVCaptureSession` or `AVCaptureDevice` or `AVCaptureMetadataOutput` to `QRScanner.Input`.

-----

Please read the CLA carefully before submitting your contribution to Mercari.
Under any circumstances, by submitting your contribution, you are deemed to accept and agree to be bound by the terms and conditions of the CLA.

https://www.mercari.com/cla/

### CODE REVIEW

File: QRScannerSample/QRScannerSample/ViewController.swift [hunk 1]
```diff
@@ -74,9 +74,7 @@ private extension ViewController {
         }
         alertController.addAction(copyAction)
         let searchWebAction = UIAlertAction(title: "Search Web", style: .default) { [weak self] _ in
-            UIApplication.shared.open(URL(string: "https://www.google.com/search?q=\(code)")!, options: [
-                :]
-                , completionHandler: nil)
+            UIApplication.shared.open(URL(string: "https://www.google.com/search?q=\(code)")!, options: [:], completionHandler: nil)
```
Comment (daichiro) [inline QRScannerSample/QRScannerSample/ViewController.swift]: Fix broken indent

File: QRScannerTests/TestCases/QRScannerViewTests.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        super.setUp()
+        qrScanner = QRScannerView()
+        delegate = QRScannerSpyDelegate()
+    }
+
+    override func tearDown() {
+        super.tearDown()
+    }
+
+    func testConfigure() {
```
Comment (daichiro) [inline QRScannerTests/TestCases/QRScannerViewTests.swift]: This test is sample.
For device simulator, `AVCaptureDevice.default(.video)` is always failed.

Of course we should test `didSuccess` and `didChangeTorchActive` so we will create mocks of `AVCaptureDevice` or so on.

File: QRScannerTests/Utils/QRScannerSpyDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    var didFailure: (QRScannerError) -> Void { get set }
+    var didSuccess: (String) -> Void { get set }
+    var didChangeTorchActive: (Bool) -> Void { get set }
+}
+
+protocol QRScannerSpyDelegateResultType: AnyObject {
+    var result: QRScannerSpyDelegateResult { get set }
+}
+
+class QRScannerSpyDelegate: QRScannerViewDelegate, QRScannerSpyDelegateResultType, QRScannerSpyDelegateResult {
```
Comment (daichiro) [inline QRScannerTests/Utils/QRScannerSpyDelegate.swift]: This is helper class for delegate.

File: QRScanner/QRScannerView.swift [hunk 4]
```diff
@@ -26,6 +26,19 @@ public extension QRScannerViewDelegate where Self: AnyObject {
...
+        public var focusImage: UIImage?
+        public var focusImagePadding: CGFloat?
+        public var animationDuration: Double?
+
+        public static var `default`: Input {
+            return .init(focusImage: nil,
+                         focusImagePadding: nil,
+                         animationDuration: nil)
+        }
+    }
```
Comment (hitsubunnu) [inline QRScanner/QRScannerView.swift]: 👍 Good idea!

**Comment by daichiro** [comment]
Thanks!

**Pull request merged**